solutions/standard/DA-keys.md

```diff
@@ -0,0 +1,65 @@
+# Defining keys in a Key Management Service configuration in an IBM Cloud Project
+
+When adding Key Management Service from the IBM Catalog to an IBM Cloud project it is possible
+to configure additional key rings and keys. This is achieved by entering edit
+mode for the project configuration and selecting the `Configure` pane and the `optional` tab.
+
+For the keys property the supporting information is "A list of objects which contain the key ring name, a flag indicating if this key ring already exists, and a flag to enable force deletion of the key ring. In addition, this object contains a list of keys with all of the information on the keys to be created in that key ring."
+
+To enter a custom value use the edit action to open the `Edit Array` pane. KMS Key Ring and Key
+configurations are added to the array here.
+
+    [
+      {
+        "key_ring_name": "da-ring-1",
+        "existing_key_ring": false,
+        "force_delete_key_ring": true,
+        "keys": [
+          {
+            "key_name": "da-key-1-1",
+            "rotation_interval_month": 1,
+            "standard_key": false,
+            "dual_auth_delete_enabled": false,
+            "force_delete": true
+          },
+          {
+            "key_name": "da-key-1-2",
+            "rotation_interval_month": 1,
+            "standard_key": false,
+            "dual_auth_delete_enabled": false,
+            "force_delete": true
+          }
+        ]
+      },
+      {
+        "key_ring_name": "da-ring-2",
+        "existing_key_ring": false,
+        "force_delete_key_ring": true,
+        "keys": [
+          {
+            "key_name": "da-key-2-1",
+            "rotation_interval_month": 1,
+            "standard_key": false,
+            "dual_auth_delete_enabled": false,
+            "force_delete": true
+          }
+        ]
+      }
+    ]
+
+This example includes all the configuration options, for two Key Rings, one containing two Keys
+and the other containing a single Key.
+
+## Key Ring options
+
+- `key_ring_name` (required) is the name choosen for the Key Ring
+- `existing_key_ring` (optional, default false) can be set to true if the Key Ring already exists and Keys should be added to it
+- `force_delete_key_ring` (optional, default true) configures the force option using destroy, or removing the configuration
+
+## Key options
+
+- `key_name` (required) is the name choosen for the Key
+- `rotation_interval_month` (optional, default 1) configures the Key rotation interval
+- `standard_key` (optional, default false) configures the Key type
+- `dual_auth_delete_enabled`  (optional, default false) configures the dual authorization for deletion option to be used during destroy
+- `force_delete` (optional, default true) configures the force option to be used during destroy
```

Comment:
I suggest that we remove this and just point to the inputs in the readme. Move it to the beginning section before the example.

> For more information, see the descriptions in the [Inputs section](/README.md#inputs) of the main readme file.

Reply:
I am not sure about this suggestion. The documentation is from a specific variable, keys, in the UI. I believe it would be confusing to link back to the inputs in the root module. The inputs in the root module are similar, but not the same as the inputs in the DA.

Reply:
OK. I'd suggest moving this before the example, and putting both sections in an H2 section.

## Options

### Key ring options

- key_ring_name: Is there more information about how a name is used? Otherwise, the description is like saying "put your name in the name field." Not particular helpful. For example, include the constraints and connect it with the [KP API](https://cloud.ibm.com/apidocs/key-protect) and CLI parameter (ID)
    - `key_ring_name`: The ID that identifies this key ring. The key ring name must be unique within an instance. The name can be up to between 2 and 100 characters and can contain spaces, letters and numbers.
- force_delete_key_ring: can we explain or point to the docs ([this?](https://cloud.ibm.com/docs/key-protect?topic=key-protect-delete-keys&interface=api#delete-keys-force-delete)) that explain more about the force option. As in 
    - `force_delete_key_ring` (optional, default = `true`) Whether to force delete the key ring with a `destroy` command or when the projects configuration is removed.  If the key ring contains keys, they are moved to the `default` key ring For more information, see [Deleting key rings](https://cloud.ibm.com/docs/key-protect?topic=key-protect-grouping-keys&interface=api#delete-key-ring-api).

### Key options

- `key_name`: A human-readable name that identifies this key. To protect your privacy, don't use personal data, such as your name or location. The key name can be between 2 nad 90 characters.
- `rotation_interval_month` (optional, default = `1`): The key rotation time interval in months, from 1 to 12.
- `standard_key` (optional, default = `false`): Whether a standard encryption key is created. For more information, see [Key types](https://cloud.ibm.com/docs/key-protect?topic=key-protect-envelope-encryption#key-types).
- `dual_auth_delete_enabled` (optional, default = `false`): Whether to use dual authorization when deleting the key. For more information, see [Using dual authorization](https://cloud.ibm.com/docs/key-protect?topic=key-protect-manage-dual-auth).
- `force_delete` (optional, default = `true`): Whether to force delete the key. For more information, see [Considerations before deleting and purging a key](https://cloud.ibm.com/docs/key-protect?topic=key-protect-delete-purge-keys#delete-purge-keys-considerations).